src/main/java/com/sourcery/gymapp/backend/workout/service/RoutineExerciseService.java

```diff
@@ -0,0 +1,59 @@
+package com.sourcery.gymapp.backend.workout.service;
+
+import com.sourcery.gymapp.backend.workout.dto.CreateRoutineExerciseDto;
+import com.sourcery.gymapp.backend.workout.dto.CreateRoutineGridExerciseDto;
+import com.sourcery.gymapp.backend.workout.dto.ResponseRoutineExerciseDto;
+import com.sourcery.gymapp.backend.workout.dto.ResponseRoutineGridExerciseDto;
+import com.sourcery.gymapp.backend.workout.mapper.RoutineExerciseMapper;
+import com.sourcery.gymapp.backend.workout.model.Routine;
+import com.sourcery.gymapp.backend.workout.model.RoutineExercise;
+import com.sourcery.gymapp.backend.workout.repository.RoutineExerciseRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class RoutineExerciseService {
+    private final RoutineService routineService;
+    private final ExerciseService exerciseService;
+    private final RoutineExerciseRepository routineExerciseRepository;
+    private final RoutineExerciseMapper routineExerciseMapper;
+
+    @Transactional
+    public CreateRoutineGridExerciseDto updateExercisesInRoutine(
```

Comment:
since we are deleting all the exercises associated with routine prior to adding the ones from the request, the proper naming would be "replaceExercisesInRoutine"